### DISCUSSION TITLE
Aliases don't work in Webstorm

### DISCUSSION THREAD

**Discussion by Rolanddoda** [quasarframework/quasar#12199] [General - Components / Directives / etc] [answered]
Hi, does anyone uses Webstorm? I have created a project using Quasar CLI but Webstorm recognizes aliases. I opened an issue on their platform, but they told me that it's a Quasar issue because it doesn't provide a webpack.config file.

**Comment by yusufkandemir** [comment]
Hi, the Quasar starter kit has;

- a `jsconfig.json` for JS projects:
https://github.com/quasarframework/quasar-starter-kit/blob/f7031bba19f6aa0966498f7277a7686734ec28f0/template/jsconfig.json#L4-L28

- a `tsconfig.json`, which uses `@quasar/app/tsconfig-preset`:
https://github.com/quasarframework/quasar/blob/15853c8cb123ec196f8c00c36207dcff736b420c/app/tsconfig-preset.json#L17-L25

Both of them have the aliases declared, also allowing you to add your own aliases.

See "Use path aliases" in [Code Style | JavaScript | Webstorm﻿](https://www.jetbrains.com/help/webstorm/settings-code-style-javascript.html#ws_js_settings_editor_code_style_imports_tab) (_a similar page exists for Typescript_):
> Always - with this option, WebStorm always uses aliases from your webpack.config.js, system.js, or **jsconfig.json** configuration file.

So, make sure you are using an up-to-date version of Webstorm, configured this setting, and have a `jsconfig.json` or `tsconfig.json`.

**Reply by Rolanddoda** [reply]
@yusufkandemir thank you for the detailed response, but I already know what you described and I use the latest version of Webstorm with "Use path aliases" to "Always".

My `jsconfig.json` is unchanged from what Quasar CLI created.

But still, Webstorm is not able to pick up aliases.

Have you tried creating a new Quasar CLI project and opening it in the latest Webstorm version?

**Reply by yusufkandemir** [reply]
You/they only pointed out `webpack.config.js`, so I took my time to explain it. I don't use Webstorm. Everything shows that all is in place, yet Webstorm doesn't work, which means it's a Webstorm issue. Have you mentioned `jsconfig.json` and all these in the ticket you created on their platform?

Just for double-checking, how/what are you trying to import where, can you share a screenshot?

**Reply by Rolanddoda** [reply]
I searched over tickets on Webstorm and indeed there is an issue in Webstorm reading `jsconfig.json` files.

[Issue URL](https://youtrack.jetbrains.com/issue/WEB-53991)

Thank you @yusufkandemir for trying to help.

**Comment by Rolanddoda** [comment, marked as answer]
The issue is related to Webstorm and not Quasar.

For more updates to that issue follow the ticket [here](https://youtrack.jetbrains.com/issue/WEB-53991)

**Comment by DTHRT** [comment]
2024 is here. I had the same problem. This help me: https://yoodima.com/blog/how-to-create-path-aliases-in-vite-and-make-them-work-with-webstorm/
